Which DCID determines Initial keys (#3862)
quicwg/base-drafts: draft-ietf-quic-transport.md

Connection ID field with an unpredictable value.  This Destination Connection ID
MUST be at least 8 bytes in length.  Until a packet is received from the server,
the client MUST use the same Destination Connection ID value on all packets in
this connection.

> The Destination Connection ID field from the first Initial packet sent by a
> client is used to determine packet protection keys for Initial packets.  These
> keys change after receiving a Retry packet; see Section 5.2 of {{QUIC-TLS}}.

Reviewer: ```suggestion
Initial packets derive their packet protection key from their Destination Connection ID field. This means that these keys will change after receiving a Retry packet; see Section 5.2 of {{QUIC-TLS}}.
```

I don't think it helps to single out the first Initial which actually isn't special.

Author: However, this suggestion is incorrect.  You can change the value of the Destination Connection ID field.  After receiving an Initial, the client sets the Destination Connection ID field to the value the server chooses in the Source Connection ID field.  This updates the value used in Initial packets (such as an ACK), but does not change the keys.

Reviewer: Ugh. In that case this text needs to be really sharpened, and perhaps we need a diagram :)

Author: [This](https://quicwg.org/base-drafts/draft-ietf-quic-tls.html#section-5.2-7) is plenty sharp in my view.

Reviewer 2: How about an additional sentence that spells this out? "The Destination Connection ID field from the first Initial packet sent by a client is used to determine packet protection keys for Initial packets. While the Destination Connection ID can change on subsequent Initial packets ({{negotiating-connection-ids}}), the packet protection keys used for Initial packets do not change. The keys change however after receiving a Retry packet; see ... "

Author: We could, but that is the point of adding the reference.

Reviewer 2: I'm fine with the reference, but if there's any question, then being explicit doesn't hurt. I'm fine either way. Separately, I noticed that Sec 5.2 in TLS says "the Initial packet". That should be "the first Initial packet from the client".
